### PR TITLE
fix(amplify_auth_cognito) AuthError instead of serialization error

### PR DESCRIPTION
**Pull request by haverchuck**
*Issue #, if available:*
#136 

*Description of changes:*
Returns 'AMPLIFY_REQUEST_MALFORMED' AuthError instead of obtuse error for serialization on null

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterSignUpRequest.kt [hunk 1]
```diff
@@ -74,17 +74,19 @@ data class FlutterSignUpRequest(val map: HashMap<String, *>) {
 
     companion object {
         fun validate(req : HashMap<String, *>?): Boolean {
-            var valid: Boolean = true;
             if (req == null) {
-                valid = false;
+                return false;
+            }
+            if (req.get("options") == null) {
```
Comment (Ashish-Nanda) [inline packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterSignUpRequest.kt:80]: It looks like options is required. Should we not add the `@required` annotation in the API as well to make it apparent to customers, or pass a default value if possible.
eg.
```
Future<SignUpResult> signUp({@required String username, @required password, @required SignUpOptions options}
```
This is in line with suggestions 1 and 2 in #136

**Comment by Ashish-Nanda** [comment]
The change looks good. 
I think one additional step as mentioned above would be to add `@required` or a default value for SignUpOptions in the customer facing API to reduce the chance of a malformed request error being thrown in the platform code.

**Comment by haverchuck** [comment]
@Ashish-Nanda Making options required would be problematic for future plugins using the interface.

**Review by Ashish-Nanda — APPROVED** [review]
Discussed offline. Looks good for now.

**Pull request merged**